### PR TITLE
ref: fetch whitelisted nft contract addresses

### LINKED ISSUE #487: Make the whitelisted contracts appearing directly through the contract
## Description 

Today we have to add a new PFP whitelisted contract on the front and on the contract, we want to do only one and the front will adapt automatically. 

## PROPOSED STEPS
- Create a function on the PFP contract to redeem whitelisted contracts
- Add this function in the front instead of checking constants

### PR DESCRIPTION
**Pull request by irisdv**
Close #487 

Also it updates how we initialize the provider in the starknetidjs context to use Alchemy.

### CODE REVIEW

File: utils/nftService.ts [hunk 1]
```diff
@@ -32,19 +32,13 @@ export const retrieveAssets = async (
 // Filter assets based on a whitelisted array of contract addresses
 export const filterAssets = (
```
Comment (fricoben) [inline utils/nftService.ts:29]: is this tested? We want all .ts file with services to be tested ideally :)

File: context/StarknetIdJsProvider.tsx [hunk 2]
```diff
@@ -29,21 +29,24 @@ export const StarknetIdJsProvider: FunctionComponent<Context> = ({
...
     return new Provider({
-      sequencer: {
-        network:
-          process.env.NEXT_PUBLIC_IS_TESTNET === "true"
-            ? constants.NetworkName.SN_GOERLI
-            : constants.NetworkName.SN_MAIN,
+      rpc: {
+        nodeUrl: `https://starknet-${
+          isTestnet ? "goerli" : "mainnet"
+        }.g.alchemy.com/v2/${process.env.NEXT_PUBLIC_ALCHEMY_KEY}`,
```
Comment (fricoben) [inline context/StarknetIdJsProvider.tsx:41]: why do you change this in this PR ?

**Review by fricoben — CHANGES_REQUESTED** [review]
Some questions :)

File: context/StarknetIdJsProvider.tsx (hunk 2 above)
Comment (irisdv) [reply context/StarknetIdJsProvider.tsx:41]: Because it's not working anymore, the sequencer does not work now. I can do a seperate PR if you prefer.

**Review by fricoben — APPROVED** [review]
lgtm

**Pull request merged**